tests/all_tests.vcxproj

```diff
@@ -120,6 +120,12 @@
       <OptimizeReferences>true</OptimizeReferences>
     </Link>
   </ItemDefinitionGroup>
+  <ItemDefinitionGroup>
+    <ClCompile Condition="'$(PlatformToolset)' != 'v140'">
+      <AdditionalOptions>/Zc:__cplusplus</AdditionalOptions>
```

Comment:
This forces VC++ to correctly define `__cplusplus`. It's good to test that this works correctly.